packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -2036,5 +2053,7 @@ if (__DEV__) {
       updateHookTypesDev();
       return updateTransition(config);
     },
+
+    unstable_avoidThisRender,
```

Comment:
Similar note here. We shouldn't allow `avoidThisRender` inside setState reducers. Should warn.